snakefile

```diff
@@ -0,0 +1,362 @@
+
+envvars:
+    "S3_SRC",
+    "GITHUB_REF",
+    "GISAID_API_ENDPOINT",
+    "GISAID_USERNAME_AND_PASSWORD",
+    "SLACK_TOKEN",
+    "FETCH"
+
+
+configfile: "snake_config.yaml"
+
+wildcard_constraints:
+    database = "gisaid|genbank"
+
+
+localrules: all_then_clean , fetch , notify_and_upload
+
+
+## defining some of the behaviour depending on 
+## which git branch we are
+
+GIT_BRANCH = ""
+SILENT = ""
+S3_DST = ''
+
+github_ref = os.environ[ "GITHUB_REF" ]
+if github_ref == "refs/heads/master" :
+    GIT_BRANCH = "master"
+    S3_DST = os.environ['S3_SRC']
+
+elif github_ref.startswith('refs/heads/') :
+    SILENT = 'yes'
+    GIT_BRANCH = github_ref[ len('refs/heads/') : ]
+    S3_DST = os.environ['S3_SRC'] + "/branch/" + GIT_BRANCH
+
+elif github_ref=='':
+    SILENT = 'yes'
+    S3_DST = os.environ['S3_SRC'] + "/tmp"
+else:
+    print("skipping ingest for ref",github_ref)
+    exit(0)
+
+print( "S3_SRC is" , os.environ['S3_SRC'] , file=sys.stderr )
+print( "S3_DST is" , S3_DST , file=sys.stderr )
+
+
+
+
+## target rule all 
+rule all_then_clean:
+    input:
+        "notify_and_upload.gisaid.mock_output.txt",
+        "notify_and_upload.genbank.mock_output.txt"
+    shell:
+        ".bin/clean"
+
+
+## target rule gisaid
+rule gisaid_then_clean:
+    input:
+        "notify_and_upload.gisaid.mock_output.txt",
+    shell:
+        ".bin/clean"
+
+## target rule genbank
+rule genbank_then_clean:
+    input:
+        "notify_and_upload.genbank.mock_output.txt",
+    shell:
+        ".bin/clean"
+
+
+
+rule fetch:
+    output:
+        "data/{database}.ndjson"
+    params:
+        s3_dst=S3_DST,
+        database = lambda wildcards : wildcards.database ,
+        slack_channel = lambda wildcards : config['slack_channel'][wildcards.database]
+    shell:
+           '''
+            if [[ "$FETCH" == 1 ]]; then
+              ./bin/fetch-from-{params.database} > {output}
+              if [[ "$branch" == master ]]; then
+
+                dst=$S3_SRC/{params.database}.ndjson.gz
+
+                src_record_count="$(wc -l < "$src")"
+                dst_record_count="$(wc -l < <(aws s3 cp --no-progress "$dst" - | gunzip -cfq))"
+                added_records="$(( src_record_count - dst_record_count ))"
+
+                msg=""
+
+                if [[ $added_records -gt 0 ]]; then
+                    msg="📈 New nCoV records (n=$added_records) found on {params.database}."
+                elif [[ $added_records -lt 0 ]]; then
+                    msg="WARNING: the new version of {params.database} has fewer records‽"
+                
+                else
+                    msg="📈 New nCoV records (n=$added_records) found on {params.database}."
+                fi
+
+                ./bin/notify-slack $msg $SLACK_TOKEN {params.slack_channel}
```

Comment:
If we are running this somewhere public where the logs can be viewed, like we do now with ingest (https://github.com/nextstrain/ncov-ingest/runs/2353020060?check_suite_focus=true#step:4:271), will people be able to see the value of the `SLACK_TOKEN` variable as it is passed as an argument to `bin/notify-slack`? 

If so, can we avoid this by using it directly from the environment variable in any bash scripts that need it like this https://github.com/nextstrain/ncov-ingest/pull/159/files#diff-3d879f93bb87b9c04449da332db37bd189cdc4b38d7703b659d126dc9e6dec03L4 ?